### PR TITLE
playonlinux: making it work at runtime

### LINKED ISSUE #81447: Package ›playonlinux‹ fails with ›ImportError: No module named wx‹
Package ›playonlinux‹ fails with ›ImportError: No module named wx‹:
```
Looking for python... 2.7.17 - wxversion(s): 3.0-gtk2
Traceback (most recent call last):
  File "/nix/store/qnnqzh9bm66lv06nsfh0a3izkq9p2g8m-playonlinux-4.3.4/share/playonlinux/python/check_python.py", line 7, in <module>
    import wx
ImportError: No module named wx
failed tests
Looking for python2.7... 2.7.17 - wxversion(s): 3.0-gtk2
Traceback (most recent call last):
  File "/nix/store/qnnqzh9bm66lv06nsfh0a3izkq9p2g8m-playonlinux-4.3.4/share/playonlinux/python/check_python.py", line 7, in <module>
    import wx
ImportError: No module named wx
failed tests
Looking for python2.6... which: no python2.6 in (/nix/store/6qqah3a5i2bkzqn5jz5v1wp8svhdgsnj-cabextract-1.9.1/bin:/nix/store/43a2f56rpbz19xq6b7jvlkqb0a7rphf9-python-2.7.17/bin:/nix/store/nxq1850ipggnab9dmz5iyhpyqqyzw63l-gettext-0.20.1/bin:/nix/store/2nzgplqdjgf6id0m4191d0cvy3ayksxk-glxinfo-8.4.0/bin:/nix/store/ljzc9wzb7qr5ml7d1w7ffm7bidqw91li-gnupg-2.2.19/bin:/nix/store/21jsvdl3gcgq67k6nw3ps7fgybr793xq-icoutils-0.32.3/bin:/nix/store/6ccarvmz5bcjil0a1axlv27vbhm2xap4-imagemagick-6.9.10-71/bin:/nix/store/6x2m55rnvbrqf72ff1ks22m375xrq26r-netcat-gnu-0.7.1/bin:/nix/store/rpi1ggnqar9jsfcrbxxb8fih4p4r60bw-p7zip-16.02/bin:/nix/store/c5m89mi2sb15p14lhhm3kq74x4rpav8i-unzip-6.0/bin:/nix/store/1ar6wvh348304grr1g6nhxiv1xw2wfp8-wget-1.20.3/bin:/nix/store/wzx8smfxa2qq0sgqn9y5d3pzsz8f39p9-wine-5.0/bin:/nix/store/zbd9c1d8y79hxlpf0fjamlhri0g0asqd-xdg-user-dirs-0.17/bin:/nix/store/6mz5bdb180qjr3wip53nsqnzlpv91pac-xterm-351/bin:/nix/store/ci44i5wns8x37bjq5n2y38alzkf03l7d-which-2.21/bin:/nix/store/94m85nahjkk8i8kblivk0j6bbq9981z5-curl-7.68.0-bin/bin:/nix/store/2zmk008jzx0shhlpy8fqimnqiqbcyjm8-jq-1.6-bin/bin:/nix/store/c0hwk342ic9vajxpzar9rsz73k4wqbhi-bash-interactive-4.4-p23/bin:/nix/store/y3bvah0wc60gv9980i21flxx4qyk3b0s-patchelf-0.9/bin:/nix/store/w24ln4ja8i3xxfy63g9hqsnyfq4kq9mj-gcc-wrapper-9.2.0/bin:/nix/store/ificps9si1nvz85f9xa7gjd9h6r5lzg6-gcc-9.2.0/bin:/nix/store/l3sgc39zdcbhx3lvlz9rmz0mv8fbmwn4-glibc-2.27-bin/bin:/nix/store/mj2g3fk1wdixjs5v34r9wzfn97g4hhxn-coreutils-8.31/bin:/nix/store/6y42d17l6v6chas70n7gf8v8ckwab90q-binutils-wrapper-2.31.1/bin:/nix/store/7bhi29ainf5rjrk7k7wyhndyskzyhsxh-binutils-2.31.1/bin:/nix/store/l3sgc39zdcbhx3lvlz9rmz0mv8fbmwn4-glibc-2.27-bin/bin:/nix/store/mj2g3fk1wdixjs5v34r9wzfn97g4hhxn-coreutils-8.31/bin:/nix/store/qnnqzh9bm66lv06nsfh0a3izkq9p2g8m-playonlinux-4.3.4/bin:/nix/store/mj2g3fk1wdixjs5v34r9wzfn97g4hhxn-coreutils-8.31/bin:/nix/store/pavi38al37g32c0vgv12ra9byhgkmivw-findutils-4.7.0/bin:/nix/store/1blbpkji6mzs91dgdkymby9fgykrffzp-diffutils-3.7/bin:/nix/store/w0fkzm5w64dwdn1afg3majsjm23wqv4b-gnused-4.7/bin:/nix/store/dk2pfccmr2hwq2iq0m0xzab0vx2ydyxb-gnugrep-3.3/bin:/nix/store/cjsf1almnicds4li6hn7m3hci7plzzf7-gawk-5.0.1/bin:/nix/store/qpgsqrxbc7910w2hrib5nqzw9k0r6ypc-gnutar-1.32/bin:/nix/store/4fqgd4hnrdcipn3x2s27ia9dw26yl4zd-gzip-1.10/bin:/nix/store/vh1xkn5sw7hqsxzz05xz1jds1b7qqglm-bzip2-1.0.6.0.1-bin/bin:/nix/store/as5612va9alwps1hlggh201iw181md8r-gnumake-4.2.1/bin:/nix/store/715fljdrspphbdk255r729zxp4w7lcgk-bash-4.4-p23/bin:/nix/store/8m502mdfky13j0c65kghiqhgfpn78kr0-patch-2.7.6/bin:/nix/store/sv1kkczmrcbx43zgnjyxyk72kkmi7f40-xz-5.2.4-bin/bin)

Looking for python2... 2.7.17 - wxversion(s): 3.0-gtk2
Traceback (most recent call last):
  File "/nix/store/qnnqzh9bm66lv06nsfh0a3izkq9p2g8m-playonlinux-4.3.4/share/playonlinux/python/check_python.py", line 7, in <module>
    import wx
ImportError: No module named wx
failed tests
Please install python before trying to run this program
```

**To Reproduce**
1. `nix-shell --pure -p playonlinux`
2. `playonlinux`

**Expected behavior**
To open playonlinux.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.1, NixOS, 20.03pre211190.a21c2fa3ea2 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(root): `"nixos-20.03pre211190.a21c2fa3ea2"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by pasqui23**
###### Motivation for this change
Fix https://github.com/NixOS/nixpkgs/issues/81447
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/playonlinux/default.nix [hunk 1]
```diff
@@ -68,15 +72,25 @@ in stdenv.mkDerivation {
...
+  ];
+
   nativeBuildInputs = [ makeWrapper ];
 
+  preBuild = ''
+    makeFlagsArray+=(PYTHON="python -m py_compile")
+  '';
+
+
   buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/playonlinux/default.nix:86]:
```suggestion
  '';

  buildInputs = [
```

**Comment by SuperSandro2000** [comment]
ping @a1russell

**Comment by a1russell** [comment]
I stopped using Nix several years ago.  I wouldn't be able to make any intelligence contribution to this PR, as I don't remember hardly any details about how Nix works. Someone else can take over maintenance of this package.

**Comment by pasqui23** [comment]
So if @a1russell allows, can I become playonlinux's mantainer?

**Comment by SuperSandro2000** [comment]
> So if @a1russell allows, can I become playonlinux's mantainer?

> Someone else can take over maintenance of this package.

You can replace him if you like.

File: pkgs/applications/misc/playonlinux/default.nix [hunk 2]
```diff
@@ -68,15 +72,25 @@ in stdenv.mkDerivation {
...
+    ./0001-fix-locale.patch
+  ];
+
   nativeBuildInputs = [ makeWrapper ];
 
+  preBuild = ''
+    makeFlagsArray+=(PYTHON="python -m py_compile")
+  '';
+
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/playonlinux/default.nix:85]:
```suggestion
  '';

```

**Pull request merged**